### ISSUE TITLE
dataclass(slots=True) accepted when targeting Python 3.9

### ISSUE THREAD

**Issue by keuj6** [astral-sh/ty#3414]
# ty bug report: dataclasses.dataclass(slots=True) accepted for Python 3.9

## Summary

ty accepts `dataclasses.dataclass(slots=True)` when checking with Python 3.9 as the target version, even though that keyword is only supported by Python 3.10+ and raises `TypeError` at import time on Python 3.9.

## Minimal Reproduction

```python
import dataclasses

@dataclasses.dataclass(frozen=True, slots=True)
class X:
    y: int
```

Run:

```bash
ty check --python-version 3.9 slots_check.py
```

## Actual Behavior

ty reports:

```text
All checks passed!
```

The same result occurs in this project with `requires-python = ">=3.9,<3.13"`, where ty should infer Python 3.9 as the minimum supported version.

## Expected Behavior

ty should emit an error for the unsupported `slots` keyword argument when the target Python version is 3.9.

On Python 3.9, importing the module fails at runtime with:

```text
TypeError: dataclass() got an unexpected keyword argument 'slots'
```

## Context

- `dataclasses.dataclass(slots=...)` was introduced in Python 3.10.
- Project target: `requires-python = ">=3.9,<3.13"`.
- CI failure observed on Python 3.9.24 during pytest collection.
- Local command used to confirm: `ty check --python-version 3.9`.
- ty version in local Docker env: `0.0.34`.

## Why This Matters

This is a false negative for Python-version compatibility. Projects supporting Python 3.9 can pass ty validation while still shipping code that crashes immediately on import under Python 3.9.

**Comment by AlexWaygood** [comment]
Thanks for the report -- this is definitely a bug. However, please consult [our AI policy](https://github.com/astral-sh/.github/blob/main/AI_POLICY.md) when submitting reports in the future.

**Comment by keuj6** [comment]
Thanks @AlexWaygood & @charliermarsh  for the prompt action. My apologies for the bot-generated report. It was my 1st time using pi agent and it fired off w/o me realising. Will be more careful in the future.
